### DISCUSSION TITLE
Do the modules in this collection support to connect to a remote SQL server host?

### DISCUSSION THREAD

**Discussion by caidingyu** [lowlydba/lowlydba.sqlserver#182] [Q&A] [answered]
Hey, I just recently found this collection with some great modules to manage MS SQL server. Reviewing a few modules and examples, I noticed that they don't have any parameter to connect to a SQL server host (like how [community.general.mssql_script module](https://docs.ansible.com/ansible/latest/collections/community/general/mssql_script_module.html) works).

Does it mean this collection is designed to only manage MS SQL server on the host which is the Ansible target and where PS module dbatools installed? If so, does it mean it can't be used to manage AWS RDS instances, for which we don't have full access to the hosts and can't install PS module or connect with Ansible directly?

To manage AWS RDS we usually need to access via a jump host. In that case it would be very helpful if the modules in this collection could support to connect to a remote host. Thanks!

**Comment by briantist** [comment, marked as answer]
The modules in this collection all take a parameter `sql_instance` for which SQL server to manage, and make a direct connection that way. They do not require PowerShell to be executed on the SQL server host itself (though that can work too).

If your inventory is SQL server instances (including RDS), you can [use `delegate_to:` to run the modules on a jump host](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_delegation.html#delegating-tasks).

The modules are also all tested with `delegate_to: localhost` (running them on the controller in `pwsh`), however they use an experimental connection and shell plugin to do so.

I helped write those plugins, and I have some loose, low priority plans to publish them but they aren't published now.

**Reply by caidingyu** [reply]
I didn't realise the `sql_instance` parameter can connect remote server. Will have a try.
Thanks very much!
